functions/private/ConvertTo-Icon.ps1

```diff
@@ -2,19 +2,75 @@ function ConvertTo-Icon {
     <#
 
         .DESCRIPTION
-        This function will convert PNG to ICO file
+        This function will convert BMP, GIF, EXIF, JPG, PNG and TIFF to ICO file
+
+        .PARAMETER bitmapPath
+        The file path to bitmap image to make '.ico' file out of.
+        Supported file types according to Microsoft Documentation is the following:
+        BMP, GIF, EXIF, JPG, PNG and TIFF.
+
+        .PARAMETER iconPath
+        The file path to write the new '.ico' resource.
+
+        .PARAMETER overrideIconFile
+        An optional boolean Parameter that makes the function overrides
+        the Icon File Path if the file exists. Defaults to $true.
+
+        .EXAMPLE
+        try {
+            ConvertTo-Icon -bitmapPath "$env:TEMP\cttlogo.png" -iconPath "$env:TEMP\cttlogo.ico"
+        } catch [System.IO.FileNotFoundException] {
+            # Handle the thrown exception here...
+        } catch {
+            # Print out the exception to console, and see what went wrong
+            Write-Host "Could not finish 'ConvertTo-Icon' due to unhandled exception"
+            Write-Host "Error: $_"
+        }
+        This Example makes a '.ico' file at "$env:TEMP\cttlogo.ico" File Path using the bitmap file
+        found in "$env:TEMP\cttlogo.png", the function overrides the '.ico' File if it's found.
+        this function will throw a FileNotFound Exception at the event of not finding the provided bitmap File Path.
```

Comment:
Could all of this be handled right inside this function to make using this as easy as invoking it ConvertTo-Icon without the need for having to build this try - catch construct every time the function is used?

Reply:
Done in https://github.com/ChrisTitusTech/winutil/pull/2242/commits/e8ca90e29f251d92be0458da98a4fd8e02dc7057 commit

There's still the case of bitmap File not being found, raising an exception is better then silently continuing IMHO, as it'll force the developer who uses this function to always check before call `ConvertTo-Icon` Function.

Also added a new Example which highlights that it's better to Check if Bitmap File exists or not, if it exists then call the `ConvertTo-Icon` Function, otherwise don't.